packages/core/src/components/TextField/Mask.jsx

```diff
@@ -84,9 +83,6 @@ function toDigits(value) {
  * @returns {Number}
  */
 function toNumber(value) {
-  if (typeof value !== 'string') return value;
-  if (!value.match(/\d/)) return undefined;
```

Comment:
I removed these 2 lines because the checks are made prior to calling this function, and because all the other helper functions (i.e. `toDigits`, `toNumber`, `toDigitsAndAsterisk`) do not make any checks either. I figured it would be easier to handle the errors in one place in `maskValue()`.